modules/imap/hm-imap.php

```diff
@@ -2134,6 +2152,12 @@ public function get_folder_list_by_level($level='') {
                 'name_parts' => $folder['name_parts'],
             );
         }
+        if ($with_subscription_state) {
+            $subscribed_folders = array_column($this->get_mailbox_list(true), 'name');
```

Comment:
For some reason, it doesn't work locally when I test. I unsubscribe from some folders but the list still shows as subscribed.

Reply:
I need to test different accounts for that. On my side It is working. The only thing a bit strange that I noticed is that when a subfolder is subscribe but not it's parent, the parent is still being shown.

Reply:
I tested with a somewhat recent postfix +dovecot imap server.